### PR TITLE
Simple auction

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by chriseth** [comment]
Will add the blind auction later.

File: _docs/blind-auction.md [hunk 1]
```diff
@@ -3,3 +3,104 @@ layout: docs
...
+    }
+
+    function () {
+        // This function gets executed if a
+        // transaction with invalid data is sent to
+        // the contract or just ether without data.
+        // We revert the send so that no-one
+        // accidentally loses money when using the
+        // contract.
+        throw;
```
Comment (CJentzsch) [inline _docs/blind-auction.md:103]: Although it is semantically easy to `throw`, it is not nice that all gas is burned. It is nicer to just return the money and do nothing else. The same is valid for the other `throw`s . Imo `throw` should really only be used for exceptional events.

Comment (chriseth) [inline _docs/blind-auction.md:103]: You are right, the fact that the gas is burned is sub-optimal, but I think this actually _is_ an exceptional event. Furthermore, "returning" the money is trivial by throwing. Returning it by using `msg.sender.send(msg.value)` might fail. In my opinion, the increased gas costs weigh less serious than accidentally losing a potentially extreme amount of ether.

Comment (CJentzsch) [inline _docs/blind-auction.md:103]: I agree. It is safer.

**Pull request merged**